synapse/storage/databases/main/event_federation.py

```diff
@@ -53,6 +53,14 @@
 logger = logging.getLogger(__name__)
 
 
+# All the info we need while iterating the DAG while backfilling
+class BackfillQueueNavigationItem(NamedTuple):
+    depth: int
+    stream_ordering: int
+    event_id: str
+    type: str
```

Comment:
Created `BackfillQueueNavigationItem` so we can create them once and get some better type hinting and assertions when trying to dip into some random fields on a `row`.

Was getting worried a typo would get introduced somewhere especially now that we have 4 fields instead of 2 (before this PR, we just fetched `depth` and `event_id`)